js/whitebit.js

```diff
@@ -615,11 +708,439 @@ module.exports = class whitebit extends Exchange {
+            request['activation_price'] = this.numberToString (this.priceToPrecision (symbol, stopPrice));
+            if (type === 'limit' || type === 'stopLimit') {
+                // it's a stop-limit-order
+                method = 'v4PrivateOPostOrderStopLimit';
+            } else if (type === 'market' || type === 'stopMarket') {
+                // it's a stop-market-order
+                method = 'v4PrivatePostOrderStopMarket';
+            }
+        } else {
+            if (type === 'market') {
+                // it's a regular market order
+                method = 'v4PrivatePostOrderMarket';
+            }
+            if (type === 'limit') {
+                // it's a regular limit order
+                method = 'v4PrivatePostOrderNew';
+            }
+        }
+        // aggregate common assignments regardless stop or not
+        if (type === 'limit' || type === 'stopLimit') {
+            if (price === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder requires a price argument for a stopLimit order');
+            }
+            const convertedPrice = this.priceToPrecision (symbol, price);
+            request['price'] = convertedPrice;
+        }
+        if (type === 'market' || type === 'stopMarket') {
+            if (side === 'buy') {
+                let cost = this.safeNumber (params, 'cost');
+                const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
+                if (createMarketBuyOrderRequiresPrice) {
+                    if (price !== undefined) {
+                        if (cost === undefined) {
+                            cost = amount * price;
+                        }
+                    } else if (cost === undefined) {
+                        throw new InvalidOrder (this.id + " createOrder() requires the price argument for market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument or in the 'cost' extra parameter (the exchange-specific behaviour)");
+                    }
+                } else {
+                    cost = (cost === undefined) ? amount : cost;
+                }
+                request['amount'] = this.costToPrecision (symbol, cost);
+            }
+        }
+        if (method === undefined) {
+            throw new ArgumentsRequired (this.id + 'Invalid type:  createOrder() requires one of the following order types: market, limit, stopLimit or stopMarket');
+        }
+        const response = await this[method] (this.extend (request, params));
+        return this.parseOrder (response);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
```

Comment:
If we require a symbol (this has to be checked against the actual exchange API by sending a request without a symbol), then we must be explicit about it, for reference: 

- https://github.com/ccxt/ccxt/blob/master/js/okex.js#L1769
- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L2703
- https://github.com/ccxt/ccxt/blob/master/js/gateio.js#L2620

Reply:
Yup I've missed that symbol was optional here